torch/utils/_sympy/value_ranges.py

```diff
@@ -83,9 +83,12 @@ def __contains__(self, x):
         x = simple_sympify(x)
         return sympy_generic_le(self.lower, x) and sympy_generic_le(x, self.upper)
 
-    def tighten(self, other: "ValueRanges"):
+    def tighten(self, other):
         """Given two ValueRanges, returns their intersection"""
-        # Some invariants
+        return self & other
+
+    # Intersection
+    def __and__(self, other):
```

Comment:
Return type for these functions should always be "ValueRange", right?

Reply:
yep, similar to `dict`.

Reply:
Might want to explicitly typehint that. :)